wagtail/admin/static_src/wagtailadmin/js/core.js

```diff
@@ -83,6 +83,57 @@ function enableDirtyFormCheck(formSelector, options) {
 }
 
 $(function() {
+    // Load icon sprite and cache it in local storage.
+    function loadIconSprite() {
+        var spriteURL = wagtailConfig.ADMIN_URLS.SPRITE;
+        var revisionKey = 'spriteRevision';
+        var dataKey = 'spriteData';
+
+        // Feature check
+        if (!document.createElementNS || !document.createElementNS('http://www.w3.org/2000/svg', 'svg').createSVGRect) {
+            return true;
+        }
+
+
+        var isLocalStorage = 'localStorage' in window && window.localStorage !== null;
+        var request;
+        var data;
+        var insertIt = function () {
+            document.body.insertAdjacentHTML('afterbegin', data);
+        }
+        var insert = function () {
+            if (document.body) insertIt();
+            else document.addEventListener('DOMContentLoaded', insertIt);
+        }
+
+        if (isLocalStorage && localStorage.getItem(revisionKey) === spriteURL) {
+            data = localStorage.getItem(dataKey);
+            if (data) {
+                insert();
+                return true;
+            }
+        }
+
+        try {
+            request = new XMLHttpRequest();
+            request.open('GET', spriteURL, true);
+            request.onload = function () {
+                if (request.status >= 200 && request.status < 400) {
+                    data = request.responseText;
+                    insert();
+                    if (isLocalStorage) {
+                        localStorage.setItem(dataKey, data);
+                        localStorage.setItem(revisionKey, spriteURL);
+                    }
+                }
+            }
+            request.send();
+        } catch (e) {
+        }
```

Comment:
Is there anything we can do if the XHR fails to provide a fallback or a notice, or are we good with just ignoring it?

Reply:
I would suggest `console.error(e);`

Reply:
What kind of errors are we anticipating here to require the try/catch? Since XHR is asynchronous, any HTTP errors resulting from the sprite fetch will happen after we've already left this try/catch block - all that will happen in that case is that the `request.onload` callback will fail to trigger. (If we need to catch those errors - and I'm not sure we do - we'd need a `request.onerror`.)